### PR TITLE
ci(safe-tools-client): Add github actions to deploy safe-tools-client

### PR DESCRIPTION
**Pull request by pcjun97**
Similar to deployment of `web-client`.
Uses native aws commands `s3 sync` and `cloudfront create-invalidation` instead of adding them to ember's `deploy.js`.

### CODE REVIEW

File: .github/actions/deploy-safe-tools-client/action.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      working-directory: packages/safe-tools-client
+      run: yarn build:safe-tools-client --environment ${{ inputs.environment }}
+
+    - name: Configure AWS credentials
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        role-to-assume: ${{ env.AWS_ROLE_ARN }}
+        aws-region: us-east-1
+
+    - name: Sync content to S3
```
Comment (lukemelia) [inline .github/actions/deploy-safe-tools-client/action.yml:49]: I would prefer to use ember-cli-deploy here. It is standard in the ember world and allows developers to add additional plugins to adjust the deploy process. Happy to help with this if you want.

**Review by lukemelia — CHANGES_REQUESTED** [review]

File: .github/actions/deploy-safe-tools-client/action.yml (hunk 1 above)
Comment (pcjun97) [reply .github/actions/deploy-safe-tools-client/action.yml]: Added the ember deploy plugins and updated the action and script.
I am unfamiliar with the tools though, so definitely appreciate the help to review them thoroughly!

File: .github/workflows/push-main.yml [hunk 2]
```diff
@@ -450,3 +456,27 @@ jobs:
...
+        with:
+          path: |
+            node_modules
+            packages/*/node_modules
+            packages/firebase-functions/*/functions/node_modules
+          key: ${{ runner.os }}-node-modules-${{ hashFiles('yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-node-modules-
+
+      - name: Deploy web-client
```
Comment (lukemelia) [inline .github/workflows/push-main.yml:478]:
```suggestion
      - name: Deploy safe-tools-client
```

File: packages/safe-tools-client/config/deploy.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    },
+    cloudfront: {
+      objectPaths: ['/*'],
+    },
+  };
+
+  if (deployTarget === 'staging') {
+    ENV.build.environment = 'production';
+    ENV.s3.bucket = 'cardstack-safe-tools-client-staging';
+    ENV.cloudfront.distribution = 'PLACEHOLDER';
```
Comment (lukemelia) [inline packages/safe-tools-client/config/deploy.js:16]: Do we know our distribution IDs and/or we make them available as environment variables to Github Actions?

**Review by lukemelia — CHANGES_REQUESTED** [review]

File: packages/safe-tools-client/config/deploy.js (hunk 3 above)
Comment (backspace) [reply packages/safe-tools-client/config/deploy.js:16]: They’re [hardcoded](https://github.com/cardstack/cardstack/blob/main/packages/web-client/config/deploy.js#L111) for web-client (and ssr-web [previews](https://github.com/cardstack/cardstack/blob/main/packages/ssr-web/config/deploy.js#L77))

**Review by lukemelia — APPROVED** [review]
Looks good! 🚀

**Pull request merged**